File: 剑指offer/9.用两个栈实现队列.py
class CQueue:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def appendTail(self, value: int) -> None:
        self.stack1.append(value)

    def deleteHead(self) -> int:
        if len(self.stack2) == 0:
            while len(self.stack1) != 0:
                self.stack2.append(self.stack1.pop(-1))

        if len(self.stack2) == 0:
            return -1
        return self.stack2.pop(-1)

File: 剑指offer/9.用两个栈实现队列.cpp
#include <stack>
using namespace std;

class CQueue {
public:
    stack<int> st1;
    stack<int> st2;

    CQueue()
    {
    }

    void appendTail(int value)
    {
        this->st1.push(value);
    }

    int deleteHead()
    {
        if (st2.empty()) {
            while (!st1.empty()) {
                st2.push(st1.top());
                st1.pop();
            }
        }

        if (st2.empty()) {
            return -1;
        }

        int ret = st2.top();
        st2.pop();

        return ret;
    }
};

/**
 * Your CQueue object will be instantiated and called as such:
 * CQueue* obj = new CQueue();
 * obj->appendTail(value);
 * int param_2 = obj->deleteHead();
 */